### DISCUSSION TITLE
Respect tsconfig path aliases when organizing imports

### DISCUSSION THREAD

**Discussion by willruggiano** [biomejs/biome#1533] [Ideas]
If I alias ./src to `@/` for example, and then import `@/whatever`, that import ideally would be organized the same as relative imports. Currently it gets pushed to the very top of the import list.

**Comment by arendjr** [comment]
With our upcoming organizer revamp, you can configure custom orders: https://next.biomejs.dev/assist/actions/organize-imports/

Does that satisfy your use case?

**Reply by siketyan** [reply]
Though custom orders can achieve the use case (for me), it still make sense to configure the orders automatically from `tsconfig.json` as a zero-config solution.

**Reply by ematipico** [reply]
> orders automatically

And what's the automatic *(default)* order for **user** aliases? :) That's definitely something that we have answer to, if we want to offer such a feature

**Reply by siketyan** [reply]
Sorry, was not clear about that. Actually I wondered if the `@/` prefix in tsconfig.json#paths can be considered as a group between external and relative imports, as it's typically used for absolute imports. However path aliases aren't only for absolute imports, and they can be external or internal imports. So we can't have meaningful default for this 😔

**Reply by siketyan** [reply]
I noticed `@/` will be considered as absolute imports by default in the new assist.

> Aliases: sources starting with @/, #, ~, or %. They usually are Node.js subpath imports or TypeScript path aliases.

I think this also fits @willruggiano's use case, so no additional configuration will be needed on Biome v2.0.

**Reply by Conaclos** [reply]
Aliases got their own section. They are placed after packages and before (absolute and relative) paths by default.

**Reply by Netail** [reply]
Mhm we use just `@` as a prefix for our import aliasses (e.g. `@components/navigation/header`, `@configs/i18n-config.ts`). We would like to place these imports under the `:PACKAGE:` imports, however they are grouped within the `:PACKAGE:` imports. A regex for `@**` won't work either as we would have to exclude every alias from the `:PACKAGE:` group first, which varies between apps & packages.

Maybe somehow read the underlying path to see if it comes from node_modules or read the tsconfig to get all the aliasses, but that would be quite complex especially with monorepos (multiple tsconfigs) & extending

**Reply by siketyan** [reply]
> Maybe somehow read the underlying path to see if it comes from node_modules or read the tsconfig to get all the aliasses, but that would be quite complex especially with monorepos (multiple tsconfigs) & extending

True. And import aliases are not only for TypeScript's path aliases, e.g. bundlers including Vite have such a feature.
